### PR TITLE
Improve grouped metadata

### PR DESCRIPTION
**Pull request by vithom**
Hello,

Here is a small improvement in the image information module to display the result of the metadata concatanation.
Capture d’écran de 2021-05-13 22-10-22
As the exemple above, it displays the value of the metadata if all selected images share it or it displays "\<various values\>" if they differ.

### CODE REVIEW

**Comment by vithom** [comment]
@TurboGit : Be careful, there will be conflicts with #8979. I don't know which one is better to integrate before.

File: src/libs/metadata_view.c [hunk 1]
```diff
@@ -339,457 +350,668 @@ static gint _lib_metadata_sort_index(gconstpointer a, gconstpointer b)
...
                                   -1, &stmt, NULL);
-      if(sqlite3_step(stmt) == SQLITE_ROW) mouse_over_id = sqlite3_column_int(stmt, 0);
+      if (sqlite3_step(stmt) == SQLITE_ROW)
+      {
+        mouse_over_id = sqlite3_column_int(stmt, 0);
+        count = sqlite3_column_int(stmt, 1);
+      }
       sqlite3_finalize(stmt);
 
       // Still -1 => no selection in progress
```
Comment (AlicVB) [inline src/libs/metadata_view.c:526]: update comment ?

**Review by AlicVB — APPROVED** [review]
Works great on my side. No real perf issues, even with a 90K images selection !
Great work !

File: src/libs/metadata_view.c (hunk 1 above)
Comment (vithom) [reply src/libs/metadata_view.c]: Yes, updated. Thanks.

**Comment by vithom** [comment]
I would have liked to update the french translation but I haven't poedit installed on my machine.
Translation for french could be "<valeurs multiples>"

File: src/libs/metadata_view.c [hunk 2]
```diff
@@ -339,457 +350,667 @@ static gint _lib_metadata_sort_index(gconstpointer a, gconstpointer b)
...
-        tooltip_parts[next_tooltip_part++] = _(flag_descriptions[2]);
-      }
+  if (!img) goto fill_minuses;
 
-      if(img->flags & DT_IMAGE_RAW)
-      {
-        value[4] = 'r';
-        tooltip_parts[next_tooltip_part++] = _(flag_descriptions[3]);
-      }
+  if (img->film_id == -1)
```
Comment (TurboGit) [inline src/libs/metadata_view.c:629]: no space between if and (

File: src/libs/metadata_view.c [hunk 3]
```diff
@@ -339,457 +350,667 @@ static gint _lib_metadata_sort_index(gconstpointer a, gconstpointer b)
...
+  {
+    dt_image_cache_read_release(darktable.image_cache, img);
+    goto fill_minuses;
+  }
 
-      if(img->flags & DT_IMAGE_HDR)
+  // Update the metadata values
+  for (int32_t md = 0; md < md_size; md++)
```
Comment (TurboGit) [inline src/libs/metadata_view.c:636]: likewise

File: src/libs/metadata_view.c [hunk 4]
```diff
@@ -339,457 +350,667 @@ static gint _lib_metadata_sort_index(gconstpointer a, gconstpointer b)
...
+  {
+    if (skip[md] == TRUE)
```
Comment (TurboGit) [inline src/libs/metadata_view.c:638]: likewise ... please checks for all occurences

File: src/libs/metadata_view.c [hunk 5]
```diff
@@ -339,457 +350,667 @@ static gint _lib_metadata_sort_index(gconstpointer a, gconstpointer b)
...
 
-      if(img->flags & DT_IMAGE_AUTO_PRESETS_APPLIED)
-      {
-        value[7] = 'a';
-        tooltip_parts[next_tooltip_part++] = _(flag_descriptions[6]);
-      }
+    char text[PATH_MAX] = {0};
 
-      if(img->flags & DT_IMAGE_NO_LEGACY_PRESETS)
+    switch (md)
```
Comment (TurboGit) [inline src/libs/metadata_view.c:657]: likewise

File: src/libs/metadata_view.c [hunk 6]
```diff
@@ -339,457 +350,667 @@ static gint _lib_metadata_sort_index(gconstpointer a, gconstpointer b)
...
-      const time_t exif_timestamp = mktime(&tt_exif);
-      _metadata_update_timestamp(md_exif_datetime, &exif_timestamp, self);
-    }
-    else
-      _metadata_update_value(md_exif_datetime, img->exif_datetime_taken, self);
+      case md_exif_focus_distance:
+        (void)g_strlcpy(text, NODATA_STRING, sizeof(text));
+        if(!(isnan(img->exif_focus_distance) || (fpclassify(img->exif_focus_distance) == FP_ZERO) ))
+        {
+          (void)g_snprintf(text, sizeof(text), "%.2f m", (double)img->exif_focus_distance);
```
Comment (TurboGit) [inline src/libs/metadata_view.c:784]: As we are at it we probably want "%.2f m" to be translatable _("%.2f m") as in Russian IINM m is M.

**Review by TurboGit — CHANGES_REQUESTED** [review]
First review, I'm going to test now.

**Review by TurboGit — COMMENTED** [review]
BTW, also we can remove the SHOW_FLAGS #define as we can now select the columns to display on the metadata view module. So this is superfluous.

**Comment by vithom** [comment]
I've corrected the remarks.

**Review by TurboGit — APPROVED** [review]
Looks good to me now. Thanks!

**Pull request merged**